File: java/src/test/java/com/adaptionsoft/games/trivia/GameTest.java
package com.adaptionsoft.games.trivia;

import static org.junit.Assert.*;

import com.adaptionsoft.games.trivia.runner.GameRunner;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.*;
import java.util.Random;
import java.util.stream.IntStream;

public class GameTest {

	@Test
	public void itsLockedDown() throws Exception {

        Random randomizer = new Random(123455);
        ByteArrayOutputStream resultStream = new ByteArrayOutputStream();
        System.setOut(new PrintStream(resultStream));

        IntStream.range(1,15).forEach(i -> GameRunner.playGame(randomizer));

        Approvals.verify(resultStream.toString());

	}
}

File: kotlin/src/test/kotlin/com/adaptionsoft/games/trivia/GameTest.kt
package com.adaptionsoft.games.trivia


import com.adaptionsoft.games.trivia.runner.GameRunner
import org.approvaltests.Approvals
import org.junit.Test

import java.io.*
import java.util.Random
import java.util.stream.IntStream

class GameTest {

    @Test
    @Throws(Exception::class)
    fun itsLockedDown() {

        val randomizer = Random(123455)
        val resultStream = ByteArrayOutputStream()
        System.setOut(PrintStream(resultStream))

        IntStream.range(1, 15).forEach { i -> GameRunner.playGame(randomizer) }

        Approvals.verify(resultStream.toString())

    }
}
